scripts/sbatch.py

```diff
@@ -0,0 +1,476 @@
+       get defaul values if not provided
+    """
+    for header in [SlurmHeader.ntasks, SlurmHeader.cpus_per_task, SlurmHeader.ntasks_per_core]:
+        if not slurm_header_dict[header.value]:
+            slurm_header_dict[header.value] = header.default
+            slurm_header_dict[header.value] = header.default
+        else:
+            slurm_header_dict[header.value] = int(slurm_header_dict[header.value])
+    #special treatment for WCT.
+    # convert str to datetime -> datetime to second
+    if not slurm_header_dict[SlurmHeader.mem_per_cpu.value]:
+        slurm_header_dict[SlurmHeader.mem_per_cpu.value] = process_mem_per_cpu(get_value_from_conf(
+            slurm_conf,
+            slurm_header_dict.partition,
+            keyword="DefMemPerCPU",
+            all2float=True,
+        ))
+    else:
+        slurm_header_dict[SlurmHeader.mem_per_cpu.value] = process_mem_per_cpu(
+            slurm_header_dict[SlurmHeader.mem_per_cpu.value]
+        )
+    #rule for share node resource
+    slurm_header_dict[SlurmHeader.exclusive.value]=shared_TRES_conf(slurm_conf, slurm_header_dict[SlurmHeader.partition.value]) 
+    return slurm_header_dict
+
+
+def get_hpc_conf(partition):
+    """Get hpc name and path to slurm.conf according to partition name in the slurm file header"""
+    for machine in [MauiPartition,MauiAncilPartition,MahuikaPartition]:
+        for p in machine:
+            if partition in p.id:
+                slurm_conf = p.config_path  
+                hpc = p.hpc
+                cpu_per_node = p.cpus
+                mem_per_node = p.mem
+            
+                return hpc, slurm_conf, cpu_per_node, mem_per_node
+    raise IndexError("no matching partition name found")
+
+
+def compare_hours_and_sbatch(requested_hours, available_hours, hpc, sl_to_sbatch, sbatch_extra_args):
+    """
+       Compare requested hour with available hours
+       and either submit the job or reject
+    """
+    if requested_hours <= available_hours:
+        cmd = "{} -M {} {} {}".format(SBATCH_BIN, hpc, sl_to_sbatch, sbatch_extra_args)
+        print(cmd)
+        subprocess.call(cmd, shell=True)
+    else:
+        sys.exit("Not enough core hours left, please apply for more")
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("slurm_to_sbatch", help="path to slurm file to be sbatched")
+    parser.add_argument('sbatch_extra_args', nargs='*', help="all args following xx.sl")
+    parser.add_argument(
+        "--core_hour_json",
+        default="core_hours.json",
+        help="path to core hour json file getting from dashboard.db, default is core_hours.json",
+    )
+    parser.add_argument(
+        "--user", default=getpass.getuser(), help="the user who's sbatching the job"
+    )
+    args = parser.parse_args()
+
+    assert os.path.exists(args.slurm_to_sbatch)
+    assert os.path.exists(args.core_hour_json)
+
+    # Get raw headers
+    header_dict = process_slurm_header(args.slurm_to_sbatch)
```

Comment:
Should be merged with sbatch_extra_args to check for variables. 
Probably don't need to check for environment variables, but worth considering

Reply:
this will introduce all new level of complexity.
since we dont use this behavior often, i would say its on low priority.
Its will be marked as a TODO.

Reply:
Agreed. I already reckon we've gone overboard with this already. We shouldn't have actually done slurm parsing as a first step.